src/scripts/Frame.js

```diff
@@ -0,0 +1,31 @@
+function Frame() {
+  this.rollsRemaining = 2;
+  this.totalScoreRecord = [];
+}
+
+Frame.prototype.registerRoll = function(pinsKnockedOver) {
+  if (pinsKnockedOver === 10) {
+    this.rollsRemaining = 0;
+  }
+  else {
+    this.rollsRemaining --;
+  }
+  this.totalScoreRecord.push(pinsKnockedOver);
+};
+
+Frame.prototype.isInProgress = function() {
+  if (this.rollsRemaining > 0) {
+    return true;
+  }
+  else {
+    return false;
+  }
+};
+
+Frame.prototype.totalScore = function() {
+  var scoreCount = 0;
```

Comment:
You can use the `reduce` method on Array, to simplify the logic and get rid of the state var